atomate/vasp/database.py

```diff
@@ -103,100 +111,57 @@ def insert_task(self, task_doc, use_gridfs=False):
-                "aeccar2",
-                "elfcar",
-            ):
-                if (
-                    vol_data_name in task_doc["calcs_reversed"][0]
-                ):  # only store idx=0 data
-                    vol_data[vol_data_name] = json.dumps(
-                        task_doc["calcs_reversed"][0][vol_data_name], cls=MontyEncoder
-                    )
-                    del task_doc["calcs_reversed"][0][vol_data_name]
 
-        # insert the task document
-        t_id = self.insert(task_doc)
+        big_data_to_store = {}
 
-        # insert the dos into gridfs and update the task document
-        if dos:
-            dos_gfs_id, compression_type = self.insert_gridfs(
-                dos, "dos_fs", task_id=t_id
-            )
-            self.collection.update_one(
-                {"task_id": t_id},
-                {"$set": {"calcs_reversed.0.dos_compression": compression_type}},
-            )
-            self.collection.update_one(
-                {"task_id": t_id}, {"$set": {"calcs_reversed.0.dos_fs_id": dos_gfs_id}}
-            )
+        def extract_from_calcs_reversed(obj_key):
+            """
+            Grab the data from calcs_reversed.0.obj_key and store on gridfs directly or some Maggma store
+            Args:
+                obj_key: Key of the data in calcs_reversed.0 to store
+            """
+            if obj_key in task_doc["calcs_reversed"][0]:
+                calcs_r_data = task_doc["calcs_reversed"][0][obj_key]
+            else:
+                calcs_r_data = task_doc["calcs_reversed"][0][obj_key]
 
-        # insert the bandstructure into gridfs and update the task documents
-        if bs:
-            bfs_gfs_id, compression_type = self.insert_gridfs(
-                bs, "bandstructure_fs", task_id=t_id
-            )
-            self.collection.update_one(
-                {"task_id": t_id},
-                {
-                    "$set": {
-                        "calcs_reversed.0.bandstructure_compression": compression_type
-                    }
-                },
-            )
-            self.collection.update_one(
-                {"task_id": t_id},
-                {"$set": {"calcs_reversed.0.bandstructure_fs_id": bfs_gfs_id}},
-            )
+            big_data_to_store[obj_key] = calcs_r_data
+
+            # remove the big object from all calcs_reversed
+            # this can catch situations were the drone added the data to more than one calc.
+            for i_calcs in range(len(task_doc["calcs_reversed"])):
+                del task_doc["calcs_reversed"][i_calcs][obj_key]
+
+        # drop the data from the task_document and keep them in a separate dictionary (big_data_to_store)
+        if self._maggma_store_type in VALID_STORES or use_gridfs and "calcs_reversed" in task_doc:
+            for data_key in OBJ_NAMES:
+                if data_key in task_doc["calcs_reversed"][0]:
+                    extract_from_calcs_reversed(data_key)
```

Comment:
I don't like this programming pattern as it is not very transparent. Can you instead make this function actually return the data and change this line to:

```python
calcs_r_data  = extract_from_calcs_reversed(data_key)
```